gen2-record-replay/libraries/depthai_record.py

```diff
@@ -242,3 +270,193 @@ def stream_out(name, fps, out, noEnc=False):
+                        "width": {"type": "integer"},
+                        "fields": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "name": {"type": "string"},
+                                    "offset": {"type": "integer"},
+                                    "datatype": {"type": "integer"},
+                                    "count": {"type": "integer"}
+                                }
+                            },
+                        },
+                        "is_bigendian": {"type": "boolean"},
+                        "point_step": {"type": "integer"},
+                        "row_step": {"type": "integer"},
+                        "data": {"type": "string", "contentEncoding": "base64"},
+                        "is_dense": {"type": "boolean"}
+                    },
+                },
+            ).encode("utf8")
+        )
+
+        # create and register channel
+        self.channel_id = self.writer.register_channel(
+            schema_id=schema_id,
+            topic="pointClouds",
+            message_encoding="json",
+        )
+
+    # send in image read with "cv2.getCvFrame"
+    def imageSave(self, img):
+        # convert cv2 image to .jpg format
+        # is_success, im_buf_arr = cv2.imencode(".jpg", img)
+
+        # read from .jpeg format to buffer of bytes
+        # byte_im = im_buf_arr.tobytes()
+        byte_im = img.tobytes()
+
+        # data must be encoded in base64
+        data = base64.b64encode(byte_im).decode("ascii")
+
+        tmpTime = time.time_ns()
+        sec = math.trunc(tmpTime / 1e9)
+        nsec = tmpTime - sec
+
+        self.writer.add_message(
+            channel_id=self.channel_id,
+            log_time=tmpTime,
+            data=json.dumps(
+                {
+                    "header": {"stamp": {"sec": sec, "nsec": nsec}},
+                    "format": "jpeg",
+                    "data": data,
+                }
+            ).encode("utf8"),
+            publish_time=tmpTime,
+        )
+
+    # send in point cloud object read with
+    # "o3d.io.read_point_cloud" or
+    # "o3d.geometry.PointCloud.create_from_depth_image"
+    # seq is just a sequence number that will be incremented in main  program (int from 0 to number at end of recording)
+    def pointCloudSave(self, pcd, seq):
+        points = np.asarray(pcd.points)
+
+        # points must be read to a buffer and then encoded with base64
+        buf = bytes()
+        for point in points:
+            buf += struct.pack('f', float(point[0]))
+            buf += struct.pack('f', float(point[1]))
+            buf += struct.pack('f', float(point[2]))
+
+        data = base64.b64encode(buf).decode("ascii")
+
+        tmpTime = time.time_ns()
+        sec = math.trunc(tmpTime / 1e9)
+        nsec = tmpTime - sec
```

Comment:
Are these variables even used?

Reply:
yes they are, they are sent in the header of the message